files/filesystem.h

```diff
@@ -0,0 +1,52 @@
+/*
+ *  filesystem.h - Filesystem operations
+ *
+ *  Copyright (C) 2020 Aleksander Miera
+ *  Copyright (C) 2000-2020  The Exult Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+#ifndef FILESYSTEM_H
+#define FILESYSTEM_H
+
+#ifdef HAVE_CONFIG_H
+#  include <config.h>
+#endif
+
+#include <system_error>
+#include <string>
+
+#if (HAVE_FILESYSTEM && USE_STDFILESYSTEM)
+#include <filesystem>
+#endif /* (HAVE_FILESYSTEM && USE_STDFILESYSTEM) */
+
+namespace fs {
+
+#if (HAVE_FILESYSTEM && USE_STDFILESYSTEM)
+using namespace std::filesystem;
```

Comment:
No `using namespace` directives in headers, please.

Reply:
Note it's just "repacking" the namespace (line 36). If you're still against it I think I can fiddle with ifdefs and use `namespace fs = std::filesystem;` instead.

Reply:
Using `namespace fs = std::filesystem;` is the better option, yes. But note that you do not even use anything from `std::filesystem` in the header, so you can just move that line to the .cc instead.

Reply:
Yeah, but note that filesystem.cc is not compiled if filesystem header from stdlib is found. But I guess I could condtionally include that in `utils.cc` instead.